chrome/content/zotero/integration/citationDialog.js

```diff
@@ -1218,6 +1249,53 @@ const IOManager = {
 		}
 		this.toggleDialogMode(desiredMode);
 	},
+
+	// Check if items added into the dialog belong to a library different from
+	// the library of other cited items. If so, show a warning message.
+	// If the user decides not to proceed, remove all items from other libraries.
+	async checkCrossLibraryCitations() {
+		// If the dialog was already shown and confirmed, do nothing
+		if (this.crossLibraryCitationsAllowed) return true;
+		// Fetch all cited items. Use io._citationsByItemIDPromise
+		// instead of io.getItems() because the later omits the items from the current field
+		let citedItemsByID = {};
+		if (io._citationsByItemIDPromise) {
+			citedItemsByID = await io._citationsByItemIDPromise;
+		}
+		let citedItems = Object.keys(citedItemsByID).map(id => Zotero.Cite.getItem(id));
+		// If the document already has cited items from multiple libraries, or not cited items at all, do nothing
+		let citedLibraries = new Set(citedItems.map(item => item.libraryID).filter(l => l));
+		if (citedLibraries.size !== 1) return true;
+		// If the items being added belong to the same library as all cited items, do nothing
+		let citedLibraryID = citedLibraries.values().next().value;
```

Comment:
Won't this return the libraryID of the newly added citation if it's being cited as the first item of the first citation in the document, i.e. being citation insert before all other citations in the doc?